dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/ValidatingEnumParamConverterProvider.java

```diff
@@ -0,0 +1,150 @@
+package io.dropwizard.jersey.validation;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
+import io.dropwizard.jersey.DropwizardResourceConfig;
+import io.dropwizard.jersey.errors.ErrorMessage;
+import org.glassfish.jersey.server.model.Resource;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.HEAD;
+import javax.ws.rs.OPTIONS;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import javax.ws.rs.ext.Provider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * Provides converters to jersey for enums used as resource parameters.
+ *
+ * <p>By default jersey will return a 404 if a resource parameter of an enum type cannot be converted. This class
+ * provides converters for all enum types used as resource parameters that provide better error handling. If an
+ * invalid value is provided for the parameter a {@code 400 Bad Request} is returned and the error message will
+ * include the parameter name and a list of valid values.</p>
+ */
+@Provider
+public class ValidatingEnumParamConverterProvider<C extends Enum<C>> implements ParamConverterProvider {
```

Comment:
No problem -- the power of open source 🎉 

To answer your questions:

- Yes it is probably prudent to make the implementations exactly the same.
- So extracting the shared code into an `Enums` class in `dropwizard-util` would be a fine idea
- And yes a rename to `FuzzyEnumParamConverterProvider`, I'm assuming, will help discover-ability